robotpy_installer/cli_deploy.py

```diff
@@ -383,6 +393,33 @@ def _ensure_requirements(
             if not python_exists:
                 logger.warning("Python is not installed on RoboRIO")
 
+        if python_exists:
+            with wrap_ssh_error("getting python version"):
+                python_version = roborio_utils.get_python3_version(ssh)
+
+            if python_version != required_pyversion:
+                python_exists = False
+                m, mn = python_version
+                rm, rmn = required_pyversion
+                python_invalid = f"python{m}{mn}"
+
+                if no_install:
+                    raise Error(
+                        "An invalid version of python3 was found on the roboRIO\n"
+                        "- could not update it because no-install was specified\n"
+                    )
```

Comment:
Currently this no_install path doesn't print the version found. It probably should.